runtime/rococo/src/bridge_messages.rs

```diff
@@ -0,0 +1,409 @@
+/// Can be computed by subtracting encoded call size from raw transaction size.
+const TX_EXTRA_BYTES: u32 = 130;
+
+/// Rococo chain as it is seen at Rococo.
+pub type RococoAtRococo = RococoLikeChain<AtRococoWithWococoMessageBridge, crate::RococoGrandpaInstance>;
+
+/// Rococo chain as it is seen at Wococo.
+pub type RococoAtWococo = RococoLikeChain<AtWococoWithRococoMessageBridge, crate::RococoGrandpaInstance>;
+
+/// Wococo chain as it is seen at Wococo.
+pub type WococoAtWococo = RococoLikeChain<AtWococoWithRococoMessageBridge, crate::WococoGrandpaInstance>;
+
+/// Wococo chain as it is seen at Rococo.
+pub type WococoAtRococo = RococoLikeChain<AtRococoWithWococoMessageBridge, crate::WococoGrandpaInstance>;
+
+/// Rococo/Wococo chain from message lane point of view.
+#[derive(RuntimeDebug, Clone, Copy)]
+pub struct RococoLikeChain<B, GI> {
+	__bridge_definition: PhantomData<B>,
+	_at_this_chain_grandpa_pallet_instance: PhantomData<GI>,
+}
+
+impl<B, GI> ChainWithMessages for RococoLikeChain<B, GI> {
+	type Hash = crate::Hash;
+	type AccountId = crate::AccountId;
+	type Signer = primitives::v1::AccountPublic;
+	type Signature = crate::Signature;
+	type Weight = Weight;
+	type Balance = crate::Balance;
+}
+
+impl<B, GI> ThisChainWithMessages for RococoLikeChain<B, GI> {
+	type Call = crate::Call;
+
+	fn is_outbound_lane_enabled(lane: &LaneId) -> bool {
+		*lane == [0, 0, 0, 0]
+	}
+
+	fn maximal_pending_messages_at_outbound_lane() -> MessageNonce {
+		MAXIMAL_PENDING_MESSAGES_AT_OUTBOUND_LANE
+	}
+
+	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight> {
+		let inbound_data_size = InboundLaneData::<crate::AccountId>::encoded_size_hint(
+			MAXIMAL_ENCODED_ACCOUNT_ID_SIZE,
+			1,
+			1,
+		)
+		.unwrap_or(u32::MAX);
+
+		MessageTransaction {
+			dispatch_weight: MAX_SINGLE_MESSAGE_DELIVERY_CONFIRMATION_TX_WEIGHT,
+			size: inbound_data_size
+				.saturating_add(EXTRA_STORAGE_PROOF_SIZE)
+				.saturating_add(TX_EXTRA_BYTES),
+		}
+	}
+
+	fn transaction_payment(transaction: MessageTransaction<Weight>) -> crate::Balance {
+		// current fee multiplier is used here
+		bridge_runtime_common::messages::transaction_payment(
+			crate::BlockWeights::get().get(frame_support::weights::DispatchClass::Normal).base_extrinsic,
+			crate::TransactionByteFee::get(),
+			pallet_transaction_payment::Pallet::<crate::Runtime>::next_fee_multiplier(),
+			|weight| crate::constants::fee::WeightToFee::calc(&weight),
+			transaction,
+		)
+	}
+}
+
+impl<B, GI> BridgedChainWithMessages for RococoLikeChain<B, GI> {
+	fn maximal_extrinsic_size() -> u32 {
+		max_extrinsic_size()
+	}
+
+	fn message_weight_limits(_message_payload: &[u8]) -> RangeInclusive<Weight> {
+		// we don't want to relay too large messages + keep reserve for future upgrades
+		let upper_limit = messages_target::maximal_incoming_message_dispatch_weight(max_extrinsic_weight());
+
+		// we're charging for payload bytes in `With?ococoMessageBridge::transaction_payment` function
```

Comment:
Is this a typo, or a shorthand for `Rococo/Wococo`? If it's a shorthand I would do something like:
```suggestion
		// we're charging for payload bytes in `With(Wococo | Rococo)MessageBridge::transaction_payment` function
```

Instead